pandas/core/window/aggregators.py

```diff
@@ -0,0 +1,159 @@
+import abc
+from functools import partial
+from typing import Optional
+
+import numpy as np
+
+from pandas._typing import Scalar
+
+
+class BaseAggregator(abc.ABC):
+    """
+    Interface to return the current value of the rolling aggregation
+    at the current step
+    """
+
+    def __init__(self, values: np.ndarray, min_periods: int) -> None:
+        self.values = values
+        self.min_periods = min_periods
+
+    @abc.abstractmethod
+    def query(self, start: int, stop: int) -> Scalar:
+        """
+        Computes the result of an aggregation for values that are between
+        the start and stop indices
+
+        Parameters
+        ----------
+        start : int
+            Current starting index
+
+        stop : int
+            Current stopping index
+
+        Returns
+        -------
+        Scalar
+            A scalar value that is the result of the aggregation.
+        """
+
+    def _meets_minimum_periods(self, values: np.ndarray) -> bool:
+        """
+        Checks that the passed values contains more non-NaN values
+        than min_periods.
+
+        Parameters
+        ----------
+        values: ndarray
+            values of the current window
+
+        Returns
+        -------
+        Bool
+        """
+        return np.count_nonzero(~np.isnan(values)) >= self.min_periods
+
+
+class AggKernel(abc.ABC):
+    """Interface that computes the aggregation value"""
+
+    def __init__(self):
```

Comment:
I would almost just remove the ``__init__`` and just define count in the kernels themselves. Otherwise you end up calling super() all the time in addition to defining things like total.